idl/read_command.idl.hh

```diff
@@ -50,6 +50,7 @@ class partition_slice {
 struct max_result_size {
     uint64_t soft_limit;
     uint64_t hard_limit;
+    uint64_t page_size [[version 4.7]] = 0;
```

Comment:
Do we need to change partition_slice for this? Isn't it enough to look at a cluster feature?

If the entire cluster supports native reversed queries, then we use the same logic for reversed queries as for forward queries.

My concern is that we have three different variables here and no one will be able to understand what is what.

Reply:
Better 3 than the 4 this started with ;-)
Anyway, if we keep 3, please do add a comment somewhere explaining what they mean, especially focusing on explaining the new version and just then how backward-compatibility is achieved.

Reply:
The thing is that despite the entire cluster having native reverse support, we can still have a reversing reader somewhere down the stack, and while the rest of the stack has to obey the rules of a normal query (use `page_size`), the reversing reader has to check the soft/hard limits instead. So its best to always send all 3 and use those that are needed on the replica.

Reply:
You mean if there is a k/l sstable involved in a reversed read? I think k/l are old enough that we can require that people upgrade their sstables if they have reversed reads (or: the reversing reader can override those numbers somehow). I don't think we should increase complexity here for that corner case.

(we should also have the cluster automatically upgrade sstables, it's about time now)

Reply:
Why would we need to change partition_slice, @avikivity ?

Reply:
I meant read_command. This is changing read_command via max_result_size.

But, I agree that having three limits makes sense - they are independent.

However, we need detailed documentation about these limits, especially since their meaning depends on a cluster feature. It's very easy to assume the wrong thing when looking at them.

Reply:
Yes. I'm about to add an extensive comment on max_result_size.